common/smoke-test/run.js

```diff
@@ -75,8 +79,15 @@ async function executeSample(sample) {
     // Set the process directory to the sample's directory because some samples
     // use file paths relative to the sample's directory.
     process.chdir(directory);
-    const entryPoint = require(`${directory}/${sampleFile}`).main;
-    await entryPoint();
+    const moduleExports = require(`${directory}/${sampleFile}`) ?? {};
+
+    for (const key of Object.keys(moduleExports)) {
+      let entryPoint = moduleExports[key];
+      if(typeof moduleExports[key] === "function") {
+        await entryPoint();
+        break;
+      }
+    }
```

Comment:
You can use `.find` to the same effect here: 

```suggestion
    const entryPoint = Object.values(moduleExports).find((v) => typeof v === "function");
    
    if (entryPoint) await entryPoint();
```